### PR TITLE
[ICDS Dashboard]Add new view for daily indicators

### PR DESCRIPTION
**Pull request by Rohit25negi**
Enable downloading daily indicators in csv format so that no manual work is there. These set of indicators are send to JS everyday.
ticket:  https://app.asana.com/0/871068296963156/1145079637410529/f

@calellowitz  @emord  @czue  @snopoke  please review this Pr. Feel free to point the nits as well if you think thats required. Also its good to do close review Since the aggregation code is touched(Though not much).

This is pretty urgent from other team perspective.

### CODE REVIEW

File: custom/icds_reports/reports/daily_indicators.py
Comment (calellowitz) [inline custom/icds_reports/reports/daily_indicators.py]: What is this code doing?

File: custom/icds_reports/views.py
Comment (calellowitz) [inline custom/icds_reports/views.py]: Is this view like the aww_inactive one, where it is just supposed to be directly hit? If so, why does it use API auth?

File: custom/icds_reports/views.py
Comment (calellowitz) [inline custom/icds_reports/views.py]: How is this view location safe? And more generally, how are you controlling access to this view?

**Review by calellowitz — CHANGES_REQUESTED** [review]
Few questions

File: custom/icds_reports/reports/daily_indicators.py
Comment (Rohit25negi) [reply custom/icds_reports/reports/daily_indicators.py]: hey this code is to fetch the data from agg_Awc_daily_view at agg_level=1. It basically start looking from yesterday's date and keep going back in time day by day until it finds the data.

File: custom/icds_reports/views.py
Comment (Rohit25negi) [reply custom/icds_reports/views.py]: Ah!! yeah Good catch. it has to be `login_and_domain_required`

File: custom/icds_reports/reports/daily_indicators.py
Comment (calellowitz) [reply custom/icds_reports/reports/daily_indicators.py]: Why are we doing that here? If the current numbers are unavailable, we dont want to silently just send yesterday's

File: custom/icds_reports/views.py
Comment (Rohit25negi) [reply custom/icds_reports/views.py]: I made it to be only accessed when user is logged in. Do you think it needs more protection as this is not present at UI to be found by users?

File: custom/icds_reports/reports/daily_indicators.py
Comment (Rohit25negi) [reply custom/icds_reports/reports/daily_indicators.py]: Its wont be silent. Filename would clearly tell its day before yesterdays data

Comment (Rohit25negi) [reply custom/icds_reports/reports/daily_indicators.py]: make the api fail when today's data is not present

File: custom/icds_reports/views.py
Comment (Rohit25negi) [reply custom/icds_reports/views.py]: Made the API accessible under feature flag

**Comment by Rohit25negi** [comment]
@calellowitz  does it look good?

**Comment by czue** [comment]
looks like cal is on this but let me know if you need a second review

File: custom/icds_reports/views.py
Comment (calellowitz) [reply custom/icds_reports/views.py]: This still isn't location safe. I would remove the decorator.

File: custom/icds_reports/reports/daily_indicators.py
Comment (calellowitz) [inline custom/icds_reports/reports/daily_indicators.py]: I don't think this should assert. I think it should return an error message, so the user can see.

**Review by calellowitz — CHANGES_REQUESTED** [review]

File: custom/icds_reports/reports/daily_indicators.py
Comment (Rohit25negi) [reply custom/icds_reports/reports/daily_indicators.py]: Hey, In the caller function. proper error message is passed to the user. do you see?

Comment (calellowitz) [reply custom/icds_reports/reports/daily_indicators.py]: I see that now, but i still think we should not be using assertions in production code. We can raise a real exception instead.

File: custom/icds_reports/views.py
Comment (Rohit25negi) [reply custom/icds_reports/views.py]: I missed it. Removed now.

File: custom/icds_reports/reports/daily_indicators.py
Comment (Rohit25negi) [reply custom/icds_reports/reports/daily_indicators.py]: I am sure about the use of assert in production if we handle it properly. Will read about it. But Made the change to raise a real exception. @calellowitz

**Review by calellowitz — APPROVED** [review]

File: custom/icds_reports/reports/daily_indicators.py
Comment (emord) [inline custom/icds_reports/reports/daily_indicators.py]: I think this would be clearer as

```python
    today = date.today()
    filename = 'CAS_Daily_Status_{}.csv'.format(today.strftime('%Y%m%d'))

    filters = {
        'aggregation_level': 1,
        'date': today - timedelta(days=1)
    }
```

File: custom/icds_reports/reports/daily_indicators.py
Comment (emord) [inline custom/icds_reports/reports/daily_indicators.py]: nit: I think it's better to use `timedelta` instead of `relativedelta` when possible. Two reasons:

1) `timedelta` is part of the standard library so it theoretically removes an external dependency (not that we could remove `dateutil` anytime soon)
2) `relativedelta` has arguments both `day` and `days` (similar for other time frames), one of which performs the arithmetic, the other replaces the value. We've had bugs come up before where these were switched unintentionally and its super hard to track down

**Review by emord — COMMENTED** [review]
My two comments are mostly just nits.

Not sure where to bring this up, but I think we should reconsider `agg_awc` as a table. Its become a very wide table and its possible that it could be split into several other tables. 

Having really wide tables can cause the amount of disk IO to be artificially inflated because each full row needs to be grabbed from disk even if a report only cares about 4 or 5 columns

File: custom/icds_reports/reports/daily_indicators.py
Comment (Rohit25negi) [reply custom/icds_reports/reports/daily_indicators.py]: Included your comments here: https://github.com/dimagi/commcare-hq/pull/25748

**Pull request merged**
